README.md

```diff
@@ -1,36 +1,312 @@
+    prod: [apns_pool_3 ],  # list of apns_pool_configs by default set to APNS production endpoint, is a list of APNS pools
+    tokens: [apns_token_1, apns_token_2 ] # optional, is a list of APNS tokens
+]
+```
+### APNS pool example
+
+Token based authentication example:
+```elixir
+apns_pool = [
+    # Token based authentication
+    auth_type: :token_based, # mandatory, :token_based or :certificate_based
+    token_id: :some_atom_id, # mandatory, token with the same id must be in `tokens` in `apns_config`
+    # Connection config
+    endpoint: "api.development.push.apple.com", # optional
+    port: 443, # optional
+    tls_opts: [], # optional
+    ping_interval: 5000, # optional
+    reconnect_attempts: 3, # optional
+    # pool config
+    tags: [:first_batch_clients, :beta_users], # optional
+    worker_num: 3, # optional
+    raw_opts: [], # optional
+]
+```
 
-If [available in Hex](https://hex.pm/docs/publish), the package can be installed
-by adding `sparrow` to your list of dependencies in `mix.exs`:
+Certificate based authentication example:
 
 ```elixir
-def deps do
-  [
-    {:sparrow, "~> 0.1.0"}
-  ]
-end
+apns_pool = [
+    # Certificate based authentication
+    auth_type: :certificate_based, # mandatory, :token_based or :certificate_based
+    cert: "path/to/apns/cert.pem", # mandatory, path to certificate file provided by APNS
+    key: "path/to/apns/key.pem", # mandatory, path to key file provided by APNS
+    # Connection config
+    endpoint: "api.push.apple.com", # optional
+    port: 443, # optional
+    tls_opts: [], # optional
+    ping_interval: 5000, # optional
+    reconnect_attempts: 3, # optional
+    # pool config
+    tags: [:another_batch_clients], # optional
+    worker_num: 3, # optional
+    raw_opts: [] # optional
+]
 ```
 
-Documentation can be generated with [ExDoc](https://github.com/elixir-lang/ex_doc)
-and published on [HexDocs](https://hexdocs.pm). Once published, the docs can
-be found at [https://hexdocs.pm/sparrow](https://hexdocs.pm/sparrow).
+### APNS token example
+
+```elixir
+apns_token = [
+          token_id: :some_atom_id, # mandatory, the same as in APNSPOOL
+          key_id: "FAKE_KEY_ID", # mandatory, data obtained form APNS account
+          team_id: "FAKE_TEAM_ID", # mandatory, data obtained form APNS account
+          p8_file_path: "path/to/file/token.p8" # mandatory, path to file storing APNS token
+        ]
+```
+
+## Include sparrow in your project
+
+```elixir
+defp deps do
+    [
+      {:sparrow, github: "esl/sprrow", tag: "cc80bbc"},
+      ]
+  end
+```
+
+## Many pools
+
+Sparrow is supporting many pools in a single `:sparrow` instance. 
+[Tags](#Tags) are used to choose pool when sending a notification.
+[](https://placehold.it/15/ff0000/ff0000?text=+) *`This feature is not yet implemented for FCM.`*
+
+## Tags
+
+Tags is a mechanism allowing to choose a pool to send a notification.
```

Comment:
Tags is a mechanism allowing to choose a pool to send a notification to?
Tags is a mechanism allowing to choose a pool to send a notification from?
Tags is a mechanism allowing to choose a pool for sending notifications?